src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -451,6 +451,111 @@ struct hotplug_offline
   }
 };
 
+struct clk_scaling_info
+{
+  using result_type = query::clk_scaling_info::result_type;
+  using data_type = query::clk_scaling_info::data_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    auto pdev = get_pcidev(device);
+
+    std::vector<std::string> stats;
+    std::string errmsg;
+    result_type ctStats;
+
+    pdev->sysfs_get("xgq_vmr", "clk_scaling_stat_raw", errmsg, stats);
+    if (!errmsg.empty()) {
```

Comment:
If I understand correctly, if there is an error message, this code is assuming that the "old XMC sysfs nodes" should be used.   

If so this is not a clean solution.  Possible alternatives:
1. See if pdev->sysfs_get() retruns a err value.  If so, use this to determine if it is the error of interest.
2. Query the driver to determine which version is support and make a decision with this value.

The solutions that involve:
A. Examine the errmsg for the error of interest. 
B. Examining if stats is empty.

Are side effect kludges that should not be done.

Reply:
> @rajkumar-xilinx please address all comments and put an ETA if possible. The VMR side is ready, but needs to be re-sync with this after the xgq_cmd_xxx header files are merged. Before we decide to switch to submodule, we still manually copy those files over to VMR. So the XRT PR goes first then VMR PR.

Hi @xdavidz I have addressed all review comments, requested Stephen to review the PR again. Thank you.